packages/headless/src/controllers/relevance-inspector/headless-relevance-inspector.ts

```diff
@@ -208,16 +234,27 @@ export function buildRelevanceInspector(
     disable() {
       dispatch(disableDebug());
     },
+
+    debugFields(fetchAll: boolean) {
```

Comment:
I'm concerned about what happen if someone pushed code to production that was calling this method with `true`? Could it open the door to maintenance cases where people report headless is slow?

If so, is there something we can do to mitigate this? e.g. logging warnings, or have it only be active for 1 request?

Alternatively, what if we only provide the `fetchFieldDescriptions` method and invest in making this easy to use (filter by kind of facet, sort, etc.)? So it becomes a two step process, where one,

1. Explores fields
2. Takes the fields you want and add them to `fieldsToInclude`

Reply:
It's similar to the debug flag (aka: when relevance inspector is enabled), which already makes headless slow (the index query execution time is much slower).

If we want to enable this only for one query, then we would need to change this for relevance-inspector as a whole. This would be a change in behaviour of the controller. Fetching all fields is just "more slowness" for queries that would be slow anyway.

Adding logging/warning is a great idea I think.

For the two steps process, I don't particularly like this since it does add more "annoying" steps than what it could be if the only thing you are interested in is leveraging those fields in your result templates, from my personal experience.

Only the field description is not enough, you also need to see what are the values of that field in relation with each individual document to be able to figure out "what it is".

The `fetchFieldsDescription()` is the thing that I wasn't sure we should add because the value it adds is lower than just being able to fetch all fields on all results while implementing an app.

Reply:
I generally think it should **not** be convenient to do the wrong thing. In this case, the trade-off is between ease of developing and performance, and honestly, leaning towards ease of development is probably the right call,

As long as there is clear, actionable logging in the console, then hopefully it will be enough for a dev/QA to realize what is happening, and be able to fix it without needing to contact Coveo support,